R/qualification-ddi.R

```diff
@@ -66,7 +66,14 @@ getQualificationDDIPlotData <- function(configurationPlan) {
           ratioList[[pkParameter]]$dose <- observedDataFrame[["Dose"]][observedDataSelection]
           ratioList[[pkParameter]]$doseUnit <- observedDataFrame[["Dose Unit"]][observedDataSelection]
           ratioList[[pkParameter]]$description <- observedDataFrame[["Description"]][observedDataSelection]
-          ratioList[[pkParameter]]$observedRatio <- observedDataFrame[[ddiPKRatioColumnName[[pkParameter]]]][observedDataSelection]
+
+          tryCatch({
+            validateIsOfType(object = observedDataFrame[[ddiPKRatioColumnName[[pkParameter]]]],"numeric")
```

Comment:
@abdullahhamadeh 
Can you document why a tryCatch checking for a numeric type of data is needed in this line ? 
(and why it is not checked in the previous lines for instance)
Is it because the column names in the DDI Ratio csv file may be ill defined by the user ?

Maybe worth creating a wrapper to extract this ratio list from the csv file / data.frame with the nomenclature for dose, unit, description defined in `reEnv` ?

Reply:
Thanks @pchelle.  For context, I am attaching the attached example of trying to read the same DDI file that is 1) comma separated and 2) semicolon separated using read.csv and read.csv2 respectively.  I get:

```
> rm(list = ls())
> dfReadBy_read.csv <- read.csv("DDI_comma_sep.csv")
> dfReadBy_read.csv2 <- read.csv2("DDI_semicolon_sep.csv")
> 
> typeof(dfReadBy_read.csv$AUCR.Avg)
[1] "double"
> typeof(dfReadBy_read.csv2$AUCR.Avg)
[1] "integer"
```

[reading-DDI.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/files/8363163/reading-DDI.zip)